src/openrct2/ride/Vehicle.cpp

```diff
@@ -8028,6 +8038,109 @@ bool Vehicle::UpdateTrackMotionForwardsGetNewTrack(uint16_t trackType, Ride* cur
     return true;
 }
 
+int32_t Vehicle::CalculateRiderBraking() const
+{
+    int32_t mode = rider_speed_preference & 0x3F;
+
+    int32_t straightTargetSpeed[] = {
+        390339, 460811, 497799, 524122, 545019, 562593, 577913,  591599,  604046,  615520,  626213,  636263,  645777,
+        654840, 663516, 671861, 679918, 687726, 695316, 702715,  709947,  717033,  723991,  730839,  737590,  744260,
+        750861, 757404, 763901, 770363, 776801, 783223, 789640,  796062,  802500,  808962,  815459,  822002,  828603,
+        835273, 842024, 848872, 855830, 862916, 870148, 877547,  885137,  892945,  901002,  909347,  918023,  927086,
+        936600, 946650, 957343, 968817, 981264, 994950, 1010270, 1027844, 1048741, 1075064, 1112052, 1182524,
```

Comment:
I don't like having the speeds defined like this but i know why you have done it. It would be useful to have the speeds in terms of mph next to it. Some of these speeds are a bit odd 5.956mph, 18.044mph. wouldn't it be neater to just go up in some simple mph increment.

Reply:
> I don't like having the speeds defined like this but i know why you have done it. It would be useful to have the speeds in terms of mph next to it.

I couldn't work out how to convert between these integer values and the actual speed.

>  Some of these speeds are a bit odd 5.956mph, 18.044mph. wouldn't it be neater to just go up in some simple mph increment.

The speeds are sampled from the inverse CDF of the normal distribution. This is done so that if speeds are selected from the table with equal probability, the distribution of rider speeds will be normal. This is done to make extreme values uncommon, because really slow riders hold everyone else up (but I don't want to have no slow values at all).

I'm still not sure if a normal distribution is the right choice, as it might be better for the distribution to be skewed towards the upper end of the range (many fast riders will get held up by the rider in front, so if you want the player to observe a range of speeds, you might need to have disproportionately more fast riders than slow ones).

Reply:
The speeds will be afair (value / 65536)_mph. (I know this is definitely correct for openloco pretty sure openrct2 is the same)

Reply:
> The speeds will be afair (value / 65536)_mph. (I know this is definitely correct for openloco pretty sure openrct2 is the same)

This doesn't seem right - in my tests, I found that 65536 seems to be closer to 2mph than 1mph. If I set the car velocity to 655360, the velocity in game is about 21mph. Are you sure the conversion isn't `value/32768`?

I can have the velocity literals in units of mph, but IMO it only really makes sense to do that if those values match up with the displayed velocity in game, and with that conversion factor they differ by a factor of 2.

Reply:
As said i haven't actually checked for openrct2 i only know openlocos off by heart.

Reply:
The actual conversion is `value / 147,456`

Reply:
> The actual conversion is `value / 147,456`

So I found [this comment](https://github.com/OpenRCT2/OpenRCT2/pull/16863#discussion_r837382056), where you said that the conversion is 65536/2.25=29127, which agrees with my experiments. I'm not sure where the 147456 came from but I think I'm going to change it to use 29127 as the conversion factor, as that actually results in sensible speeds.

Reply:
Sorry yeah multiplied it instead of divide was doing things quickly